File: StudyAsist-Android/app/src/main/java/com/studyasist/data/local/entity/Result.kt
package com.studyasist.data.local.entity

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "results",
    foreignKeys = [
        ForeignKey(
            entity = Attempt::class,
            parentColumns = ["id"],
            childColumns = ["attemptId"],
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [Index(value = ["attemptId"])]
)
data class Result(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val attemptId: Long,
    val score: Float,
    val maxScore: Float,
    val percent: Float,
    val detailsJson: String,
    val manualFeedback: String? = null
)

File: StudyAsist-iOS/StudyAsist/Models/Result.swift
//
//  Result.swift
//  StudyAsist
//

import Foundation
import SwiftData

@Model
final class Result {
    @Attribute(.unique) var id: Int64
    var attemptId: Int64
    var score: Float
    var maxScore: Float
    var percent: Float
    var detailsJson: String
    var manualFeedback: String?
    
    init(
        id: Int64 = 0,
        attemptId: Int64,
        score: Float,
        maxScore: Float,
        percent: Float,
        detailsJson: String,
        manualFeedback: String? = nil
    ) {
        self.id = id
        self.attemptId = attemptId
        self.score = score
        self.maxScore = maxScore
        self.percent = percent
        self.detailsJson = detailsJson
        self.manualFeedback = manualFeedback
    }
}
